en/docs/how-tos/overriding-endpoint-information/overriding-endpoints-for-developer-first-approach-apis.md

```diff
@@ -2,19 +2,46 @@
 
 The following sections describe how you can override endpoint information for APIs that have been added to WSO2 API Microgateway using an OpenAPI definition.
 
-### Overriding a single endpoint at runtime
+You need to define endpoints as endpoint objects in the OpenAPI definition in order to override endpoints at runtime. See [defining endpoint object]({{base_path}}/how-tos/defining-endpoints-in-an-openapi-definition/#endpoint-object).
+
+## Overriding a single endpoint
+
+#### Overriding a single endpoint at runtime
 
 Use the following command to start WSO2 Microgateway. Here the system variables are provided at server start-up to override the endpoint of the API, which has been added to WSO2 API Microgateway using an OpenAPI definition.
 
 ``` tab="Format"
-bash gateway <endpoint-name>_<endpoint-type>_endpoint_<endpoint-index>="<endpoint_URL>" -e <endpoint-name>_<endpoint-type>_basic_password=<password> <path-to-MGW-executable-file>
+gateway <path-to-MGW-executable-file> --<endpoint-name>_<endpoint-type>_endpoint_<endpoint-index>="<endpoint_URL>"
 
+- <path-to-MGW-executable-file> - Enter the path to the WSO2 API Microgateway project's executable file ( `.jar` ).
 - <endpoint-name> - Name specified in the OpenAPI definition under the `x-wso2-endpoints` OpenAPI extension.
 - <endpoint-type> -  Use one of the following values based on the type of the endpoint.
-- prod - Use this for a production endpoint.
-- sand - Use this for a sandbox endpoint.
+    - prod - Use this for a production endpoint.
+    - sand - Use this for a sandbox endpoint.
+- <endpoint-index>- The endpoint index starts from 0. Therefore, when overriding a single endpoint this value is 0.
+
+
+```
+
+``` java tab="Example"
+gateway /home/user/petstore-project/target/petstore-project.jar --myEndpoint_prod_endpoint_0="http://wso2.com" 
+```
+
```

Comment:
Shall we remove the extra new lines?

Reply:
This is because to keep the existing formatting as it is